File: client-android/lib/src/main/kotlin/stasis/client_android/lib/discovery/ClientDiscoveryAttributes.kt
package stasis.client_android.lib.discovery

import stasis.client_android.lib.model.core.NodeId
import stasis.client_android.lib.model.server.devices.DeviceId
import stasis.client_android.lib.model.server.users.UserId

data class ClientDiscoveryAttributes(
    val user: UserId,
    val device: DeviceId,
    val node: NodeId
) : ServiceDiscoveryClient.Attributes {
    override fun asServiceDiscoveryRequest(isInitialRequest: Boolean): ServiceDiscoveryRequest =
        ServiceDiscoveryRequest(
            isInitialRequest = isInitialRequest,
            attributes = mapOf(
                "user" to user.toString(),
                "device" to device.toString(),
                "node" to node.toString()
            )
        )
}

File: client-ios/lib/Sources/StasisClientLib/Discovery/ClientDiscoveryAttributes.swift
import Foundation

public struct ClientDiscoveryAttributes: Sendable, Equatable, Hashable, ServiceDiscoveryClientAttributes {
    public let user: UserId
    public let device: DeviceId
    public let node: NodeId

    public init(user: UserId, device: DeviceId, node: NodeId) {
        self.user = user
        self.device = device
        self.node = node
    }

    public func asServiceDiscoveryRequest(isInitialRequest: Bool) -> ServiceDiscoveryRequest {
        ServiceDiscoveryRequest(
            isInitialRequest: isInitialRequest,
            attributes: [
                "user": user.uuidString.lowercased(),
                "device": device.uuidString.lowercased(),
                "node": node.uuidString.lowercased()
            ]
        )
    }
}
